### ISSUE TITLE
Building Adobe Analytics SDK with Expo 51, RN 0.74 with Expo Modules, Native File changes

### ISSUE THREAD

**Issue by sainjay** [expo/expo#29744]
### Minimal reproducible example

https://github.com/sainjay/expo51-adobe-analytics

### What platform(s) does this occur on?

iOS

### Did you reproduce this issue in a development build?

Yes

### Summary

I'm trying to implement Adobe Analytics SDK in React Native 0.74.2 and Expo 51.0.11. The Adobe Analytics SDK doesn't support Expo yet, but there a some issues pointing to alternatives.

Library - https://github.com/adobe/aepsdk-react-native

https://github.com/adobe/aepsdk-react-native/issues/346

Made the following changes in documentation for AppDelegate.mm
![image](https://github.com/expo/expo/assets/136019884/844c9274-7e51-4bef-bc63-f5a075cf4327)

Added the flags in Xcode project as mentioned in this issue:
https://github.com/adobe/aepsdk-react-native/issues/247#

Added this in Podfile:
`
installer.pods_project.targets.each do |t|
      if t.name.start_with?("AEP")
        t.build_configurations.each do |bc|
            bc.build_settings['OTHER_SWIFT_FLAGS'] = '$(inherited) -no-verify-emitted-module-interface'
        end
      end
    end
`

if the import for `#import <Expo/Expo.h>`, running into these errors:
![image](https://github.com/expo/expo/assets/136019884/7bf472ee-d777-4731-b55b-36632e73d333)

But if I comment line3 and add the `#import <ExpoModulesCore/EXAppDelegateWrapper.h>`, i'm able to build:
![image](https://github.com/expo/expo/assets/136019884/1e3c168c-badb-4cbe-b881-6e5ef1f67e90)

Can you guide on what could be going wrong with Expo.h, will commenting that and using the line 4 for ExpoModulesCore is good solution in this case? Looking for guidance and information to build this properly for Expo 51. I'm exploring ExpoModules and Expo Config pluggin currently to automate the editing of native files in ios/ folder.

### Environment

expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.5
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.16.0 - ~/.nvm/versions/node/v18.16.0/bin/node
      Yarn: 3.6.1 - ~/.nvm/versions/node/v18.16.0/bin/yarn
      npm: 9.5.1 - ~/.nvm/versions/node/v18.16.0/bin/npm
      Watchman: 2024.05.06.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.15.2 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.2, iOS 17.2, macOS 14.2, tvOS 17.2, visionOS 1.0, watchOS 10.2
      Android SDK:
        API Levels: 28, 30, 31, 32, 33, 34
        Build Tools: 30.0.3, 33.0.0, 33.0.1, 34.0.0
        System Images: android-30 | Google Play ARM 64 v8a, android-31 | Google APIs ARM 64 v8a, android-34 | Google APIs ARM 64 v8a
    IDEs:
      Android Studio: 2023.2 AI-232.10227.8.2321.11479570
      Xcode: 15.2/15C500b - /usr/bin/xcodebuild
    npmPackages:
      expo: ~51.0.14 => 51.0.14 
      expo-router: ~3.5.16 => 3.5.16 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.74.2 => 0.74.2 
      react-native-web: ~0.19.10 => 0.19.12 
    Expo Workflow: bare

### Expo Doctor Diagnostics

✔ Check Expo config for common issues
✔ Check package.json for common issues
✔ Check dependencies for packages that should not be installed directly
(node:64239) Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification.
(Use `node --trace-warnings ...` to show where the warning was created)
✔ Check for issues with metro config
✖ Check for common project setup issues
✔ Check npm/ yarn versions
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check for legacy global CLI installed locally
✔ Check that native modules do not use incompatible support packages
✔ Check native tooling versions
✔ Check that native modules use compatible support package versions for installed Expo SDK
✔ Check that packages match versions required by installed Expo SDK

Detailed check results:

This project has native project folders but also has config plugins, indicating it is configured to use Prebuild. EAS Build will not sync your native configuration if the ios or android folders are present. Add these folders to your .gitignore file if you intend to use prebuild (aka "managed" workflow).

One or more checks failed, indicating possible issues with the project.

**Comment by sainjay** [comment]
`#import <Expo/Expo.h>` in AppDelegate.h is giving build errors for Adobe Analytics SDK.

**Comment by filipef101** [comment]
@sainjay Any luck?

**Comment by sainjay** [comment]
yes, created a whole config plugin to handle all the changes required for the Adobe Analytics to work. Refer below:
https://docs.expo.dev/config-plugins/plugins-and-mods/

> @sainjay Any luck?
